proxy/proxy.go

```diff
@@ -1443,6 +1444,9 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		r.URL.Path = rfc.PatchPath(r.URL.Path, r.URL.RawPath)
 	}
 
+	// Change /foo/../bar to /bar for matching and passing upstream
+	r.URL.Path = httppath.Clean(r.URL.Path)
```

Comment:
this would override the rfc.PatchPath() call above

Reply:
good catch. Yeah we need to apply it early as the pathPath might be swapping a / for a %2F and we probably want to operate on the / version. Though the critical thing for me is the proxied upstream path and the path used in matcher do not diverge